src/js/experimental/media-chrome-select-menu.js

```diff
@@ -78,14 +118,50 @@ class MediaChromeMenuButton extends window.HTMLElement {
 
     if (!this.#listboxSlot.hidden) {
       this.#listbox.focus();
+      this.#updateMenuPosition();
     } else {
       this.#button.focus();
     }
   }
 
+  #updateMenuPosition() {
+    // if the menu is hidden, skip updating the menu position
+    if (this.#listbox.offsetWidth === 0) return;
+
+    const buttonRect = this.#button.getBoundingClientRect();
+
+    // if we're outside of the controller,
+    // one of the components should have a media-controller attribute
+    // or the button wasn't slotted
+    if (
+      this.hasAttribute('media-controller') ||
+      this.#button.hasAttribute('media-controller') ||
+      this.#listbox.hasAttribute('media-controller') ||
+      !this.#buttonSlotted
+    ) {
+      this.#listbox.style.zIndex = '1';
+      this.#listbox.style.bottom = 'unset';
+      this.#listbox.style.top = buttonRect.height + 'px'
+      return;
+    }
+
+    // Get the element that enforces the bounds for the list boxes.
+    const bounds =
+      (this.getAttribute('bounds')
+        ? closestComposedNode(this, `#${this.getAttribute('bounds')}`)
+        : this.parentElement) ?? this;
+
+    const boundsRect = bounds.getBoundingClientRect();
+    const listboxRect = this.#listbox.getBoundingClientRect();
+    let position = -Math.max(buttonRect.x + listboxRect.width - boundsRect.right, 0);
+
+
+    this.#listbox.style.transform = `translateX(${position}px)`;
+  }
+
   #toggleExpanded() {
-    this.#button.setAttribute('aria-expanded', this.#expanded);
     this.#expanded = !this.#expanded;
+    this.#button.setAttribute('aria-expanded', this.#expanded);
   }
 
   enable() {
```

Comment:
the lines below in enable / disable mutating `this.#button.handleClick` should also be removed I think.
we can't make assumptions this is a media-chrome button.
can make this work with preventing the click event on the button with adding an event listener w/ use capture and stopping propagation

Reply:
Yeah, you're right, we definitely shouldn't assume that it's a media-chrome-button.

Reply:
switched to an event handler.